### PR TITLE
Add support for email on push integration.

### PR DESCRIPTION
**Pull request by ttaxon**
This adds support for the Emails on push integration.

### CODE REVIEW

File: src/main/java/org/gitlab4j/api/ServicesApi.java [hunk 1]
```diff
@@ -493,4 +494,58 @@ public void deleteCustomIssueTrackerService(Object projectIdOrPath) throws GitLa
         delete(Response.Status.OK, null, "projects", getProjectIdOrPath(projectIdOrPath), "services", "custom-issue-tracker");
 
     }
+
+    /**
+     * Get the Custom Issue Tracker service settings for a project.
```
Comment (jabby) [inline src/main/java/org/gitlab4j/api/ServicesApi.java:499]: Can you update javadoc accordingly to gitlab documentation? https://docs.gitlab.com/ee/api/services.html#emails-on-push

File: src/main/java/org/gitlab4j/api/ServicesApi.java [hunk 2]
```diff
@@ -493,4 +494,58 @@ public void deleteCustomIssueTrackerService(Object projectIdOrPath) throws GitLa
...
+     * @return a ExternalWikiService instance holding the External Wiki service settings
+     * @throws GitLabApiException if any exception occurs
+     */
+    public EmailOnPushService getEmailOnPushService(Object projectIdOrPath) throws GitLabApiException {
+        Response response = this.get(Response.Status.OK, null, "projects", getProjectIdOrPath(projectIdOrPath), "services", "emails-on-push");
+        return (response.readEntity(EmailOnPushService.class));
+    }
+
+    /**
+     * Updates the Custom Issue Tracker service settings for a project.
```
Comment (jabby) [inline src/main/java/org/gitlab4j/api/ServicesApi.java:513]: Can you update javadoc accordingly to gitlab documentation? https://docs.gitlab.com/ee/api/services.html#emails-on-push

File: src/main/java/org/gitlab4j/api/ServicesApi.java [hunk 3]
```diff
@@ -493,4 +494,58 @@ public void deleteCustomIssueTrackerService(Object projectIdOrPath) throws GitLa
...
+     * @throws GitLabApiException if any exception occurs
+     */
+    public EmailOnPushService updateEmailOnPushService(Object projectIdOrPath,  EmailOnPushService customIssueTracker) throws GitLabApiException {
+        GitLabApiForm formData = customIssueTracker.servicePropertiesForm();
+        Response response = put(Response.Status.OK, formData.asMap(), "projects", getProjectIdOrPath(projectIdOrPath), "services", "emails-on-push");
+        return (response.readEntity(EmailOnPushService.class));
+    }
+
+    /**
+     * Deletes the Custom Issue Tracker service for a project.
```
Comment (jabby) [inline src/main/java/org/gitlab4j/api/ServicesApi.java:539]: Can you update javadoc accordingly to gitlab documentation? https://docs.gitlab.com/ee/api/services.html#emails-on-push

File: src/main/java/org/gitlab4j/api/services/EmailOnPushService.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+import org.gitlab4j.api.GitLabApiForm;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+public class EmailOnPushService extends NotificationService {
+
+    public static final String RECIPIENT_PROP = "recipients";
+    public static final String DISABLE_DIFFS_PROP = "disable_diffs";
+    public static final String SEND_FROM_COMMITTER_EMAIL = "send_from_committer_email";
```
Comment (jabby) [inline src/main/java/org/gitlab4j/api/services/EmailOnPushService.java:11]: `SEND_FROM_COMMITTER_EMAIL` should be named `SEND_FROM_COMMITTER_EMAIL_PROP`

File: src/main/java/org/gitlab4j/api/services/EmailOnPushService.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    public static final String BRANCHES_TO_BE_NOTIFIED = "branches_to_be_notified";
```
Comment (jabby) [inline src/main/java/org/gitlab4j/api/services/EmailOnPushService.java:12]: `BRANCHES_TO_BE_NOTIFIED` should be named `BRANCHES_TO_BE_NOTIFIED_PROP`.

File: src/main/java/org/gitlab4j/api/services/EmailOnPushService.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        return withTagPushEvents(pushEvents, this);
+    }
+
+    @JsonIgnore
+	public String getRecipients() {
+		return ((String)getProperty(RECIPIENT_PROP));
+	}
+	public void setRecipients(String recipients) {
+		setProperty(RECIPIENT_PROP, recipients);
+	}
```
Comment (jabby) [inline src/main/java/org/gitlab4j/api/services/EmailOnPushService.java:39]: Please fix the identation here.

**Review by jabby — CHANGES_REQUESTED** [review]
Thanks for this PR. I didn't try your patch yet. 
Here are few remarks. Can you please take care of them?

**Comment by ttaxon** [comment]
All edits have been made.

**Comment by jabby** [comment]
Thanks @ttaxon 
I will provide some automated tests in another PR

**Pull request merged**